src/modules/didIpfsHelper.js

```diff
@@ -138,26 +141,46 @@ class ChluIPFSDID {
     }
 
     async publish(did, waitForReplication = true) {
-        let publicDidDocument
+        let publicDidDocument, privateKeyBase58, signature
         if (did) { 
             publicDidDocument = did.publicDidDocument
+            signature = did.signature
+            privateKeyBase58 = did.privateKeyBase58
```

Comment:
Wherever this method is called from and here too, we should nullify the private key, just for paranoia. Maybe we can deal with this during a security audit, but just noting a potential todo here.

Reply:
That's a good point. Maybe we should open an issue about private key security.

Remember js uses references for objects, does not copy them, so nullifying has to be done after the signature is made.

Reply:
Yup. There are nuances there. We should open a github card called "Known security fixes", so we can track them and be ready when we are asked about being production ready.